app/src/main/java/com/example/moviewithfragments/request/MovieApiClient.kt

```diff
@@ -0,0 +1,141 @@
+package com.example.moviewithfragments.request
+
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import com.example.moviewithfragments.AppExecutors
+import com.example.moviewithfragments.models.MovieModel
+import com.example.moviewithfragments.response.MovieSearchResponse
+import com.example.moviewithfragments.utils.Credentials
+import retrofit2.Call
+import retrofit2.Response
+import java.io.IOException
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit
+
+class MovieApiClient private constructor() {
+
+    var appExecutors = AppExecutors()
+    private val viewModelMovies: MutableLiveData<List<MovieModel>?>
+    var retrieveMoviesRunnable: RetrieveMoviesRunnable? = null
+    val movies: MutableLiveData<List<MovieModel>?>
+        get() = viewModelMovies
+
+    companion object {
+        var instance: MovieApiClient? = null
+            get() {
+                if (field == null) {
+                    field = MovieApiClient()
+                }
+                return field
+            }
+            private set
+    }
+
+    init {
+        viewModelMovies = MutableLiveData()
+    }
+
+    fun searchMoviesApi(query: String, year: Int, pageNumber: Int) {
+
+        if (retrieveMoviesRunnable != null) {
+            retrieveMoviesRunnable = null
+        }
+        val retrieveMoviesRunnable = RetrieveMoviesRunnable(query, year, pageNumber)
+        val retroFitHandler: Future<*> =
+            appExecutors.getInstance()?.getNetworkIO()!!.submit(retrieveMoviesRunnable)
+        appExecutors.getInstance()!!.getNetworkIO()?.schedule(Runnable {
```

Comment:
One more recommendation is
Instead of creating our own executors, you can use coroutines which makes it easier to handle concurrent and also parallel execution